ydb/library/yql/providers/yt/provider/yql_yt_physical_optimize.cpp

```diff
@@ -3400,6 +3400,13 @@ class TYtPhysicalOptProposalTransformer : public TOptimizeTransformerBase {
             return {};
         }
         TYtOutTableInfo outTable(outItemType, State_->Configuration->UseNativeYtTypes.Get().GetOrElse(DEFAULT_USE_NATIVE_YT_TYPES) ? NTCF_ALL : NTCF_NONE);
+
+        {
+            auto path = write.Table().Name().StringValue();
+            auto commitEpoch = TEpochInfo::Parse(write.Table().CommitEpoch().Ref()).GetOrElse(0);
+            auto dstRowSpec = State_->TablesData->GetTable(cluster, path, commitEpoch).RowSpec;
+            outTable.RowSpec->SetColumnOrder(dstRowSpec->GetColumnOrder());
```

Comment:
Column order was previously lost here, so second and further INSERTs into the same table failed